client/lib/analytics/utils/signup_domain_origin.js

```diff
@@ -0,0 +1,21 @@
+export const SIGNUP_DOMAIN_ORIGIN = {
+	use_your_domain: 'use your domain',
+	choose_later: 'choose later',
+	free: 'free',
+	custom: 'custom',
+	not_set: 'not set',
+};
+
+const SIGNUP_DOMAIN_ORIGIN_KEY = 'SIGNUP_DOMAIN_ORIGIN';
+
+export function getDomainOrigin() {
+	return window.localStorage.getItem( SIGNUP_DOMAIN_ORIGIN_KEY ) || SIGNUP_DOMAIN_ORIGIN.not_set;
+}
+
+export function setDomainOrigin( origin ) {
+	window.localStorage.setItem( SIGNUP_DOMAIN_ORIGIN_KEY, origin );
+}
+
+export function removeDomainOrigin() {
+	window.localStorage.removeItem( SIGNUP_DOMAIN_ORIGIN_KEY );
+}
```

Comment:
This file can grow exponentially with data being accessed across various areas of the codebase via getter setters in multiple locations.

The signup framework already has the dependency store to funnel data across the steps.  The data then can be extracted somewhere as shown below and then fed into the `recordSignupComplete` function.

https://github.com/Automattic/wp-calypso/blob/6f87983e61d1dd9b5f4dd8a057ffcb41ef426b72/client/signup/main.jsx#L481-L503